prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -39,66 +34,29 @@ class BlobAccess(
              * @throws MalformedURLException if the blob URL is malformed
              */
             fun getBlobFilename(blobUrl: String): String {
-                return if (blobUrl.isNotBlank())
-                    FilenameUtils.getName(URL(URLDecoder.decode(blobUrl, Charset.defaultCharset())).path) else ""
+                return if (blobUrl.isNotBlank()) {
+                    FilenameUtils.getName(URL(URLDecoder.decode(blobUrl, Charset.defaultCharset())).path)
+                } else ""
             }
         }
     }
 
     /**
-     * Upload the [report] to the blob store using the [action] to determine a folder as needed.  A [subfolderName]
-     * is optional and is added as a prefix to the blob filename.
+     * Upload the [report] to the blob store using the [action] to determine a folder as needed.
+     * A [subfolderName] is optional and is added as a prefix to the blob filename.
      * @return the information about the uploaded blob
      */
-    fun generateBodyAndUploadReport(
+    fun uploadReport(
```

Comment:
I think this function can be moved to the companion object

Reply:
Doing this requires a couple changes, may be better to save for later